spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -199,11 +206,23 @@ public Object next() {
 					return new FileMarker(filePath, Mark.END);
 				}
 				try {
-					return bufferedReader.readLine();
+					if (this.line == null && !this.done) {
+						this.line = bufferedReader.readLine();
+					}
```

Comment:
This looks for me like a hack for those cases when `hasNext()` isn't invoked. If that's OK from `Iterator` implementation perspective, we need to take care about `eof` here in the `next()`. Otherwise we miss it just after the `else` below. Moreover even with `NoSuchElementException`.
Would you mind to explain the logic?
Thanks